src/jit/ByteCodeParser.cpp

```diff
@@ -1899,6 +1899,66 @@ static void compileFunction(JITCompiler* compiler)
 #endif /* SLJIT_CONFIG_X86 */
             break;
         }
+#if defined(ENABLE_EXTENDED_FEATURES)
+        case ByteCode::I32AtomicLoadOpcode:
+        case ByteCode::I32AtomicLoad8UOpcode:
+        case ByteCode::I32AtomicLoad16UOpcode: {
+            info = Instruction::kIs32Bit;
+            requiredInit = OTLoadI32;
+            FALLTHROUGH;
+        }
+        case ByteCode::I64AtomicLoadOpcode:
+        case ByteCode::I64AtomicLoad8UOpcode:
+        case ByteCode::I64AtomicLoad16UOpcode:
+        case ByteCode::I64AtomicLoad32UOpcode: {
+            group = Instruction::Atomic;
+
+            Instruction* instr = compiler->append(byteCode, group, opcode, 1, 1);
+#if defined(WALRUS_32)
+            if (!(info & Instruction::kIs32Bit)) {
+                info = Instruction::kIsCallback;
+            }
```

Comment:
Only `I64AtomicLoadOpcode` needs a callback. The rest stores <= 32 bit, so they are supported on 32 bit systems.